code/datums/quirks/_quirk.dm

```diff
@@ -33,11 +33,25 @@
 	/// The base weight for the each quirk's mail goodies list to be selected is 5
 	/// then the item selected is determined by pick(selected_quirk.mail_goodies)
 	var/list/mail_goodies
+	/// The minimum stat where this quirk can process (if it has QUIRK_PROCESSES)
+	/// If null, then it will process regardless of stat.
+	var/minimum_process_stat = HARD_CRIT
+	/// A list of additional signals to register with update_process()
+	var/list/process_update_signals
+	/// A list of traits that should stop this quirk from processing.
+	/// Signals for adding and removing this trait will automatically be added to `process_update_signals`.
+	var/list/no_process_traits
+
+/datum/quirk/New()
+	. = ..()
+	if(no_process_traits)
+		LAZYINITLIST(process_update_signals)
+		for(var/trait in no_process_traits)
+			process_update_signals += list(SIGNAL_ADDTRAIT(trait), SIGNAL_REMOVETRAIT(trait))
```

Comment:
you only need the for loop.
for(_ in null) is a noop

Reply:
fair, it was mostly to avoid the unneeded `LAZYINITLIST` - but I can just do `LAZYADD`, looking back